tests/wandb_run_test.py

```diff
@@ -758,3 +773,45 @@ def test_init_with_settings(live_mock_server, test_settings):
         == wandb_sdk.wandb_settings.Source.INIT
     )
     run.finish()
+
+
+def test_repo_job_creation(live_mock_server, test_settings):
+    with open("test.py", "w") as f:
+        f.write('print("test")')
+    test_settings.update({"enable_job_creation": True})
+    with mock.patch.dict("os.environ", WANDB_PROGRAM="test_program.py"):
+        run = wandb.init(settings=test_settings)
+        run.finish()
+        ctx = live_mock_server.get_ctx()
+        artifact_name = list(ctx["artifacts"].keys())[0]
+
+        assert artifact_name == wandb.util.make_artifact_name_safe(
+            f"job-{run._settings.git_remote}_{run._settings.program}"
+        )
+
+
+def test_artifact_job_creation(live_mock_server, test_settings):
+    with open("test.py", "w") as f:
+        f.write('print("test")')
```

Comment:
is this safe? should we have some filesystem isolation?

Reply:
This is just cargo culted from other tests in the same file to make log_code work. This one could be removed potentially, I'll test to make sure

Reply:
hmmm... i don't think you need to remove it... 
i was just wondering if you should use a filesystem isolation context manager

Reply:
The `local_settings` fixture creates an isolated filesystem.

Reply:
ah that's cool... i didn't see it in the fixture! good to know!